### PR TITLE
fix(oracle): use timestamp of message to set final price

### PR DESCRIPTION
**Pull request by leonz789**
## Description

Set timestamp of final price with the timestamp from message
----

Closes #XXX

## Summary by CodeRabbit

- **New Features**
	- Added a new error message `ErrPriceProposalFormatInvalid` for improved error handling of price proposal message formats.

- **Bug Fixes**
	- Improved timestamp validation to ensure accurate and valid timestamp entries in messages.

- **Enhancements**
	- Updated timestamp field initializations to use consistent and meaningful values instead of hardcoded strings.
	- Introduced validation for timestamps to prevent future dates and ensure correct formatting.

### CODE REVIEW

File: x/oracle/keeper/msg_server_create_price.go [hunk 1]
```diff
@@ -52,3 +60,24 @@
 
 	return &types.MsgCreatePriceResponse{}, nil
 }
+
+func checkTimestamp(_ context.Context, msg *types.MsgCreatePrice) error {
+	now := time.Now().UTC()
```
[the remark this answers is not in the document]
Comment (MaxMustermann2) [reply x/oracle/keeper/msg_server_create_price.go]: @leonz789 why not use the block time here?

File: x/oracle/types/errors.go [hunk 2]
```diff
@@ -8,7 +8,8 @@ import (
 
 // x/oracle module sentinel errors
 var (
-	ErrSample               = sdkerrors.Register(ModuleName, 1100, "sample error")
-	ErrInvalidMsg           = sdkerrors.Register(ModuleName, 1, "invalid input create price")
-	ErrPriceProposalIgnored = sdkerrors.Register(ModuleName, 2, "price proposal ignored")
+	ErrSample                     = sdkerrors.Register(ModuleName, 1100, "sample error")
```
Comment (TimmyExogenous) [inline x/oracle/types/errors.go]: Please change the error code to zero.

File: x/oracle/keeper/msg_server_create_price.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (leonz789) [reply x/oracle/keeper/msg_server_create_price.go]: Yes, it should be block time. updated

File: x/oracle/types/errors.go (hunk 2 above)
Comment (leonz789) [reply x/oracle/types/errors.go]: updated

Comment (MaxMustermann2) [reply x/oracle/types/errors.go]: In the Cosmos sdk error documentation, it is advised that error codes start from, and include 2.

> The only restrictions on error codes are the following:
> - Must be greater than one, as a code value of one is reserved for internal errors.
> - Must be unique within the module.

https://docs.cosmos.network/main/build/building-modules/errors

EVMOS does this by using `iota + 2` as the starting var.

**Review by MaxMustermann2 — APPROVED** [review]

File: x/oracle/keeper/msg_server_create_price.go [hunk 3]
```diff
@@ -2,22 +2,30 @@ package keeper
...
 	"context"
+	"errors"
 	"strconv"
+	"time"
 
 	"github.com/ExocoreNetwork/exocore/x/oracle/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+const layout = "2006-01-02 15:04:05"
```
Comment (bwhour) [inline x/oracle/keeper/msg_server_create_price.go:13]: can we move this const timelayout into oralce/types ? 
the latter one variable  t also use this format ? 

>` const t = "2024-05-01 01:01:01"`

why not use one ?

Comment (leonz789) [reply x/oracle/keeper/msg_server_create_price.go:13]: That is required by golang. The layout is a special sequence that Go uses to understand how to interpret and format date and time strings. Random string will cause error.

**Review by TimmyExogenous — APPROVED** [review]

**Comment by cloud8little** [comment]
Test passed, the timestamp is added to the price info.

```
exocored q oracle show-prices 1
prices:
  next_round_id: "4"
  price_list:
  - decimal: 0
    price: ""
    round_id: "0"
    timestamp: ""
  - decimal: 18
    price: "318966000000"
    round_id: "1"
    timestamp: "2024-07-04 07:28:57"
  - decimal: 18
    price: "318966000000"
    round_id: "2"
    timestamp: "2024-07-04 07:28:57"
  - decimal: 18
    price: "318966000000"
    round_id: "3"
    timestamp: "2024-07-04 07:28:57"
  token_id: "1"
```

**Review by cloud8little — APPROVED** [review]

**Pull request merged**